### ISSUE TITLE
NoMethodError: undefined method `table_name' with Rails 7.1.0

### ISSUE THREAD

**Issue by wrburgess** [activerecord-hackery/ransack#1444]
Recently upgraded a rails application from `7.0.8` to `7.1.0`, using ruby `3.2.2`

Search using ransack returns this error:

```
NoMethodError: undefined method `table_name' for #<Arel::Table:0x00007f62299aa050 @name="titles",

return nil unless attr && attr.valid?
--
15 | name         = attr.arel_attribute.name.to_s
16 | table        = attr.arel_attribute.relation.table_name
17 | schema_cache = self.klass.connection.schema_cache
18 | unless schema_cache.send(:data_source_exists?, table)
```

Reverting back to rails `7.0.8` corrects the issue

**Comment by OutlawAndy** [comment]
I just hit this also.  Not sure what happened to `Arel::Table#table_name` but changing line 16 to `table        = attr.arel_attribute.relation.name` resolves the issue.

Is there a pattern in place already for handling API changes between Rails versions?  I'd be happy to put together a PR with a little guidance from someone more familiar with the project.  Thanks.

**Comment by dmitrue** [comment]
temporary solution mentioned here #1420

**Comment by deivid-rodriguez** [comment]
Closed by #1439.
